### ISSUE TITLE
Multi source apps don't contain resolved revision in .status field

### ISSUE THREAD

**Issue by jimmyjones2** [argoproj/argo-cd#13859]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

The `.status` fields of a multi source Application doesn't contain the resolved git revision in the status field, only the targetRevision. I think there need to be some way of indicating what git revision targetRevision was resolved to, so plugins, or someone inspecting the Application CRD can discover where the values came from. Otherwise will be hard to debug if updates from git have been applied by ArgoCD?

**To Reproduce**

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: guestbook2
spec:
  destination:
    namespace: default
    server: https://kubernetes.default.svc
  project: default
  sources:
  - chart: prometheus
    helm:
      valueFiles:
      - $values/helm-guestbook/values.yaml
    repoURL: https://prometheus-community.github.io/helm-charts
    targetRevision: 15.7.1
  - ref: values
    repoURL: https://github.com/argoproj/argocd-example-apps.git
    targetRevision: HEAD
  - chart: elasticsearch
    repoURL: https://helm.elastic.co
    targetRevision: 8.5.1
status:
  health:
    status: Progressing
  history:
  - deployStartedAt: "2023-05-31T20:01:02Z"
    deployedAt: "2023-05-31T20:01:04Z"
    id: 0
    revisions:
    - 8.5.1
    source:
      repoURL: ""
    sources:
    - chart: prometheus
      helm:
        valueFiles:
        - $values/helm-guestbook/values.yaml
      repoURL: https://prometheus-community.github.io/helm-charts
      targetRevision: 15.7.1
    - ref: values
      repoURL: https://github.com/argoproj/argocd-example-apps.git
      targetRevision: HEAD
    - chart: elasticsearch
      repoURL: https://helm.elastic.co
      targetRevision: 8.5.1
  operationState:
    finishedAt: "2023-05-31T20:01:04Z"
    message: successfully synced (all tasks run)
    operation:
      initiatedBy:
        username: admin
      retry: {}
      sync:
        revision: 15.7.1
        syncStrategy:
          hook: {}
    phase: Succeeded
    startedAt: "2023-05-31T20:01:02Z"
    syncResult:
      resources:
      <snip>
      revision: ""
      revisions:
      - 8.5.1
      source:
        repoURL: ""
      sources:
      - chart: prometheus
        helm:
          valueFiles:
          - $values/helm-guestbook/values.yaml
        repoURL: https://prometheus-community.github.io/helm-charts
        targetRevision: 15.7.1
      - ref: values
        repoURL: https://github.com/argoproj/argocd-example-apps.git
        targetRevision: HEAD
      - chart: elasticsearch
        repoURL: https://helm.elastic.co
        targetRevision: 8.5.1
  reconciledAt: "2023-05-31T20:01:05Z"
  resources:
    <snip>
  sourceTypes:
  - Helm
  summary:
    images:
    <snip>
  sync:
    comparedTo:
      destination:
        namespace: default
        server: https://kubernetes.default.svc
      source:
        repoURL: ""
      sources:
      - chart: prometheus
        helm:
          valueFiles:
          - $values/helm-guestbook/values.yaml
        repoURL: https://prometheus-community.github.io/helm-charts
        targetRevision: 15.7.1
      - ref: values
        repoURL: https://github.com/argoproj/argocd-example-apps.git
        targetRevision: HEAD
      - chart: elasticsearch
        repoURL: https://helm.elastic.co
        targetRevision: 8.5.1
    revisions:
    - 8.5.1
    status: Synced
```

**Expected behavior**

I'd expect the revisions of all three sources to appear in order:
```yaml
    revisions:
    - 15.7.1
    - 53e28ff20cc530b9ada2173fbbd64d48338583ba
    - 8.5.1
```

**Version**

```json
{
    "Version": "v2.6.7+5bcd846"
}
```

**Comment by JorTurFer** [comment]
I'll open a PR later on, I have noticed where is the problem

**Comment by crenshaw-dev** [comment]
The fix will be in 2.7.5 and 2.6.9 tomorrow. Thanks @JorTurFer!
